test/blackbox-tests/test-cases/github3046/run.t

```diff
@@ -17,7 +17,7 @@ are given as paramters
 
   $ dune init lib foo --ppx="foo bar"
   dune: option `--ppx': invalid element in list (`foo bar'): expected a valid
-        dune atom
+        dune 'ppx' name
   Usage: dune init [OPTION]... INIT_KIND NAME [PATH]
   Try `dune init --help' or `dune --help' for more information.
   [1]
```

Comment:
I notice that we don't have tests ensuring that invalid strings trigger errors when passed as `ppx`s or `libs` (which is why we're not catching the potential regression the current changes introduce). We should probably add that in. That doesn't have to be part of this PR, but this note should help me remember in the future :)

Reply:
Indeed, it would be helpful if we can validate other fields too. 

This PR is focused mainly on enhancing the ```dune init exe hello1 --libs="lwt,lwt.unix"``` type of dune usage/workflow.